SimpleITK v2.1.1 requires Lua 5.3, which is not available in the easybuild foss/2021a toolchain.

This patch allows the code generation during the configuration step to use Lua 5.4. 
(Lua wrapping has been disabled in the existing SimpleITK easyconfigs, anyway.)

Author: https://github.com/crubb

diff -ur SimpleITK-2.1.1/CMake/sitkGenerateFilterSource.cmake SimpleITK-2.1.1-Lua54/CMake/sitkGenerateFilterSource.cmake
--- SimpleITK-2.1.1/CMake/sitkGenerateFilterSource.cmake	2021-09-09 20:30:08.000000000 +0000
+++ SimpleITK-2.1.1-Lua54/CMake/sitkGenerateFilterSource.cmake	2022-02-05 20:01:47.594257463 +0000
@@ -7,7 +7,7 @@
   get_property( SAVE_LUA_EXECUTABLE_DOCSTRING CACHE LUA_EXECUTABLE PROPERTY HELPSTRING )
   unset(LUA_EXECUTABLE CACHE)
 
-  find_package( LuaInterp 5.3  REQUIRED )
+  find_package( LuaInterp 5.4  REQUIRED )
   set( SimpleITK_LUA_EXECUTABLE ${LUA_EXECUTABLE} CACHE PATH "Lua executable used for code generation." )
 
   if (DEFINED SAVE_LUA_EXECUTABLE)
@@ -46,10 +46,10 @@
 
 if( SITK_LUA_VERSION_RESULT_VARIABLE
       OR
-    NOT ${SimpleITK_LUA_EXECUTABLE_VERSION} VERSION_GREATER "5.2"
+    NOT ${SimpleITK_LUA_EXECUTABLE_VERSION} VERSION_GREATER "5.3"
       OR
-    NOT ${SimpleITK_LUA_EXECUTABLE_VERSION} VERSION_LESS "5.4" )
-  message(SEND_ERROR "Lua version 5.3 is required for SimpleITK_LUA_EXECUTABLE_VERSION.")
+    NOT ${SimpleITK_LUA_EXECUTABLE_VERSION} VERSION_LESS "5.5" )
+  message(SEND_ERROR "Lua version 5.4 is required for SimpleITK_LUA_EXECUTABLE_VERSION.")
 endif()
 
 # Sets "out_var" variable name to the value in the json path specified
